e2e/tests-dfx/canister_logs.bash

```diff
@@ -65,3 +65,26 @@ dfx_canister_logs_tail_n_1() {
   assert_not_contains "Alice"
   assert_contains "Bob"
 }
+
+@test "canister logs only visible to allowed viewers." {
+  install_asset logs
+  dfx_start
+  dfx canister create --all
+  dfx build
+  dfx canister install e2e_project
+  dfx canister call e2e_project hello Alice
+  sleep 2
+
+  assert_command dfx canister logs e2e_project
+  assert_contains "Hello, Alice!"
+
+  # Create identity for viewers.
+  assert_command dfx identity new --storage-mode plaintext alice
```

Comment:
Maybe it makes sense to create Alice identity first, without adding it to allowed viewers, verify it has no access to logs, then add it as a viewer and verify it now has access.

Reply:
Fixed at https://github.com/dfinity/sdk/pull/3921/commits/201e2b727e6f000da193c446a0d68d4914eb3b0c.